### PR TITLE
Added Signup Button and UI

### PR DESCRIPTION
**Pull request by jjojan**

### CODE REVIEW

File: app/src/main/java/com/example/harvestfresh/LoginActivity.java [hunk 1]
```diff
@@ -21,15 +22,22 @@ public class LoginActivity extends AppCompatActivity {
     private EditText etUsername;
     private EditText etPassword;
```
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/LoginActivity.java:23]: I believe Kal had a suggestion for renaming this that got missed from the previous PR. Will be good to address them here.

https://github.com/jjojan/HarvestFresh_Capstone/pull/7#discussion_r902960820

File: app/src/main/java/com/example/harvestfresh/SignupActivity.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.signup_activity);
+
+        btnFinalSignup = findViewById(R.id.btnFinalSignup);
+        etSignupUsername = findViewById(R.id.etSignupUsername);
+        etSignupPassword = findViewById(R.id.etSignupPassword);
+        etConfirmPassword = findViewById(R.id.etConfirmPassword);
```
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/SignupActivity.java:33]: Kal's suggestion applies here too.

File: app/src/main/java/com/example/harvestfresh/SignupActivity.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+        btnFinalSignup.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String newUsername = etSignupUsername.getText().toString();
+                String newPassword = etSignupPassword.getText().toString();
+                String confirmPassword = etConfirmPassword.getText().toString();
+
+                if(newUsername.isEmpty() || newPassword.isEmpty() || confirmPassword.isEmpty()){
+                    Toast.makeText(SignupActivity.this, "One or more of the fields are blank", Toast.LENGTH_SHORT).show();
```
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/SignupActivity.java:44]: Extract to resources xml file.

File: app/src/main/java/com/example/harvestfresh/SignupActivity.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                }
+                else if (newPassword == confirmPassword){
+                    signupUser(newUsername, newPassword);
+                }
+                else{
+                    Toast.makeText(SignupActivity.this, "Passwords do not match", Toast.LENGTH_SHORT).show();
```
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/SignupActivity.java:50]: Extract to resources xml file.

File: app/src/main/java/com/example/harvestfresh/SignupActivity.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+        user.setUsername(newUsername);
+        user.setPassword(newPassword);
+
+        user.signUpInBackground(new SignUpCallback() {
+            public void done(ParseException e) {
+                if (e == null) {
+                    goMainActivity();
+                } else {
+                    Log.e(TAG, "Error with signup");
```
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/SignupActivity.java:67]: nit: Can be a little more detailed.
```
Error - Signup failed with ParseException.
```

In general, it is good practice to also include the `exception.toString()` value in your error log message.
```
Log.e("Error - Signup failed with ParseException" + e.toString());
```

**Review by rahul-pl — COMMENTED** [review]
Great progress Jefrin. Move to the PR model is looking good.
For UI changes, like this one, I would also recommend adding a screenshot/video of the UI/Flow in the commit/PR.

File: app/src/main/java/com/example/harvestfresh/SignupActivity.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+import com.parse.ParseUser;
+import com.parse.SignUpCallback;
+
+public class SignupActivity extends AppCompatActivity {
+
+    private Button btnFinalSignup;
+    private EditText etSignupUsername;
+    private EditText etSignupPassword;
+    private EditText etConfirmPassword;
+    private static final String TAG = "SIGNUPACTIVITY";
```
Comment (kalalsherif) [inline app/src/main/java/com/example/harvestfresh/SignupActivity.java:23]: Its generally a good idea to seperate out lists of instantiations like these to make it clear where the private instance end and private static variables begin, so I would put a new line before this. Also, we generally try to sort class variables and methods from most to least important: statics before instance, and public before protected before private, so I would put TAG at the top. Following this convention makes it a lot easier to find code later, and to read through the class later and get a top-down understanding without jumping all over it.

File: app/src/main/java/com/example/harvestfresh/SignupActivity.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+
+        btnFinalSignup.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String newUsername = etSignupUsername.getText().toString();
+                String newPassword = etSignupPassword.getText().toString();
+                String confirmPassword = etConfirmPassword.getText().toString();
+
+                if(newUsername.isEmpty() || newPassword.isEmpty() || confirmPassword.isEmpty()){
```
Comment (kalalsherif) [inline app/src/main/java/com/example/harvestfresh/SignupActivity.java:43]: Nit: For long, repetitive lists of code like this, it looks a bit cleaner to put it on multiple lines:
if(newUsername.isEmpty() || 
   newPassword.isEmpty() || 
   confirmPassword.isEmpty()) {
   
   
   Also, I notice that sometimes you're putting a space before the open bracket, like on line 17 36 here, and sometimes, you don't, such as here on 43 or 46 below. Pick one convention and stick with it (I prefer space before bracket personally)

File: app/src/main/java/com/example/harvestfresh/SignupActivity.java (hunk 3 above)
Comment (jjojan) [reply app/src/main/java/com/example/harvestfresh/SignupActivity.java:44]: Sorry when you say extract into a xml are you the if statement should just have one variables whos value is determined  inside the resource file?

File: app/src/main/java/com/example/harvestfresh/SignupActivity.java (hunk 4 above)
Comment (jjojan) [reply app/src/main/java/com/example/harvestfresh/SignupActivity.java:50]: ^ same question about which part & how it should be extracted ? Should the message inside the toast be extracted elsewhere?

**Pull request merged**